conformance/lang/expressions/checking-expression/checkpanic_expr.balt

```diff
@@ -0,0 +1,1050 @@
+        error-type, function-defn, function-call-expr, optional-type
+
+function init() {
+    error? x1 = file(); // @output 10
+}
+
+class File {
+    int i = 10;
+}
+
+function file() returns error? {
+    File f = checkpanic new File();
+    io:println(f.i);
+}
+
+Test-Case: error
+Description: Test the static type of the checkpanic-expr when it is used with new expressions.
+Labels: int, error-type, function-defn, function-call-expr, optional-type, module-class-defn, DecimalNumber
+
+function init() {
+    error? x1 = file(); // @error expected '(int|error)' but found 'File'
+}
+
+class File {
+    int i = 10;
+}
+
+function file() returns error? {
+    int _ = checkpanic new File();
+}
+
+Test-Case: output
+Description: Test checkpanic-expr with field-access, optional-field-access and member-access expressions.
+Labels: array-type, boolean, decimal, DecimalFloatingPointNumber, DecimalNumber, field-access-expr, float,
+        json, list-constructor-expr, function-defn, function-call-expr, optional-type,
+        error-type, mapping-constructor-expr, string, optional-field-access-expr, checkpanic, unary-not
+
+function init() {
+    error? x1 = fieldAccess(); // @output 2
+                               // @output 2
+                               // @output abc
+                               // @output abc
+                               // @output [true,false,false]
+                               // @output [true,false,false]
+                               // @output 21
+                               // @output 21
+    io:println(x1); // @output key 'b1' not found in JSON mapping
+
+    error? x2 = memberAccess(); // @output 2
+                                // @output abc
+                                // @output [true,false,false]
+                                // @output [true,false,false]
+                                // @output 1
+
+    io:println(x2); // @output JSON value is not a mapping
+}
+
+type Rec record {
+    int x;
+    string y;
+    boolean[3] z;
+};
+
+Rec rec = {x: 2, y: "abc", z: [true, false, !true], "a": []};
+
+function fieldAccess() returns error? {
+    json j1 = {a: 21, b: true};
+
+    int i1 = checkpanic rec.x;
+    io:println(i1);
+    int i2 = checkpanic rec?.x;
+    io:println(i2);
+
+    string s1  =checkpanic rec.y;
+    io:println(s1);
+    string s2  =checkpanic rec?.y;
+    io:println(s2);
+
+    boolean[] b1 = checkpanic rec.z;
+    io:println(b1);
+    boolean[] b2 = checkpanic rec?.z;
+    io:println(b2);
+
+    int i3 = checkpanic j1.a;
+    io:println(i3);
+    int i4 = checkpanic j1?.a;
+    io:println(i4);
+
+    boolean b3 = checkpanic j1.b1;
+    io:println(b3);
+    boolean b4 = checkpanic j1?.b1;
+    io:println(b4);
+
+    string _ = checkpanic j1.c;
+}
+
+function memberAccess() returns error? {
+    json[] j1 = [1, 2, "a", 1.0, [1, 2, 3]];
+
+    int i1 = checkpanic rec["x"];
+    io:println(i1);
+
+    string s1  =checkpanic rec["y"];
```

Comment:
```suggestion
    string s1  = checkpanic rec["y"];
```

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-spec/commit/9bbe838a3d2b202e3159bf66cca24c931438633a and https://github.com/ballerina-platform/ballerina-spec/commit/13778f9e3e433f968993a7f7eacf616d851656f1